meson.build

```diff
@@ -0,0 +1,848 @@
+	'wchar.h',
+	'wctype.h',
+]
+
+check_functions = [
+	['fnmatch', '#include <fnmatch.h>'],
+	['ftruncate', '#include <unistd.h>'],
+	['gnu_get_libc_version', '#include<gnu/libc-version.h>'],
+	['isblank', '#include <ctype.h>'],
+	['mbrtowc', '#include <wchar.h>'],
+	['memcpy',  '#include <string.h>'],
+	['mkstemp', '#include <stdlib.h>'],
+	['regcomp', '#include <regex.h>'],
+	['socket', '#include <sys/socket.h>'],
+	['strerror', '#include <string.h>'],
+	['strstr', '#include <string.h>'],
+	['wcrtomb', '#include <wchar.h>'],
+	['wcscoll', '#include <wchar.h>'],
+]
+
+foreach h : check_headers
+	define = 'HAVE_' + h.underscorify().to_upper()
+	if cc.has_header(h)
+		cdata.set(define, 1)
+	else
+		cdata.set(define, false)
+	endif
+endforeach
+
+foreach f : check_functions
+	define = 'HAVE_' + f.get(0).underscorify().to_upper()
+	if cc.has_function(f.get(0), prefix : f.get(1))
+		cdata.set(define, 1)
+	else
+		cdata.set(define, false)
+	endif
+endforeach
+
+cdata.set_quoted('PACKAGE', 'geany')
+cdata.set_quoted('PACKAGE_BUGREPORT', 'https://github.com/geany/geany/issues')
+cdata.set_quoted('PACKAGE_NAME','Geany')
+cdata.set_quoted('PACKAGE_STRING', 'Geany ' + meson.project_version())
+cdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+cdata.set('GETTEXT_PACKAGE', 'PACKAGE')
+cdata.set('VERSION', meson.project_version())
+cdata.set('ENABLE_NLS', 1)
+if (host_machine.system() == 'windows')
+	cdata.set('HAVE_VTE', false)
+else
+	cdata.set('HAVE_VTE', get_option('vte'))
+endif
+cdata.set('HAVE_PLUGINS', get_option('plugins'))
+cdata.set('HAVE_SOCKET', get_option('socket'))
+if (host_machine.system() == 'windows')
+	cdata.set('GEANY_DATA_DIR', join_paths(prefix, 'data'))
+	cdata.set('GEANY_DOC_DIR', join_paths(prefix))
+else
+	cdata.set('GEANY_DATA_DIR', join_paths(prefix, get_option('datadir'), 'geany'))
+	cdata.set('GEANY_DOC_DIR', join_paths(prefix, get_option('datadir'), 'doc', 'geany'))
+endif
+cdata.set('top_srcdir', meson.source_root())
+cdata.set('top_builddir', meson.build_root())
+
+# for geany.pc (adapted from GTK+)
+pcconf = cdata
+pcconf.set('DEPENDENCIES', deps_for_pc)
+pcconf.set('prefix', get_option('prefix'))
+pcconf.set('libdir', join_paths('${exec_prefix}', get_option('libdir')))
+pcconf.set('includedir', join_paths('${prefix}', get_option('includedir')))
+pcconf.set('datarootdir', join_paths('${prefix}', get_option('datadir')))
+# actually constant, just match autconf
+pcconf.set('exec_prefix', '${prefix}')
+pcconf.set('datadir', '${datarootdir}')
+pcconf.set('localedir', join_paths('${datarootdir}', 'locale'))
+
+# needed programs
+sh = find_program('sh')
+cp = find_program('cp')
+ln = find_program('ln')
+python = find_program('python3')
+# These two are truly optional
+rst2html = find_program('rst2html', required: false)
+rst2pdf = find_program('rst2pdf', required: false)
+git = find_program('git', required: false)
+if git.found()
+	ret = run_command(git, 'rev-parse', '--short', '--revs-only', 'HEAD', check: false)
```

Comment:
I think you probably don't want this to be `check: false` because in the bizarro case that git errors out here you don't want this to be set to an empty string, though you might indeed want to check that the filesystem entry `.git` exists rather than assuming the current directory is a git checkout simply because git is installed.

The empty string is not the same as your current fallback of `-1`.

Reply:
fixed